src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -221,6 +244,7 @@ public static class ResolverUtility
         /// <param name="logger">logger</param>
         /// <param name="cancellationToken">cancellation token</param>
         /// <returns>The requested range and remote match.</returns>
+        [Obsolete("This method is marked as Obsolete. Use an overloaded method that accepts RemoteWalkContext as parameter instead.")]
```

Comment:
I searched entire `github` and there are no references to this method except in the NuGet Client code. There could be repos that are not maintained in `github` but using this public API hence marked this method as `obsolete`.

Reply:
Feels like we're overexposing the walk context API unnecessarily. 

This is equivalent to the feedback in Erick's PR for packages.config namespaces, where we passed down PackageNamespacesConfiguration instead of ISettings.

Reply:
I reverted this change from this PR. I will revisit this change in my follow-up for package namespaces.